include/nanobind/nb_lib.h

```diff
@@ -192,6 +192,10 @@ NB_CORE PyObject **seq_get(PyObject *seq, size_t *size,
 NB_CORE PyObject *capsule_new(const void *ptr,
                               void (*free)(void *) noexcept) noexcept;
 
+/// Create a new capsule object with a name
```

Comment:
Makes sense, done!

When I added the extra function I probably didn't realize `capsule_new()` was in `details::`, and was trying to avoid breaking API changes.